x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -508,9 +517,17 @@ public boolean isActive() {
     public boolean checkFeature(Feature feature) {
         boolean allowed = isAllowed(feature);
         LongAccumulator maxEpochAccumulator = lastUsed.get(feature);
+        long now = System.currentTimeMillis();
         if (maxEpochAccumulator != null) {
             maxEpochAccumulator.accumulate(epochMillisProvider.getAsLong());
         }
+
+        if (feature.minimumOperationMode.compareTo(OperationMode.BASIC) > 0 && isLicenseExpiring(now)) {
+            HeaderWarning.addWarning("Your license will expire in [{}] days. " +
```

Comment:
If we're in the grace period, shouldn't this say `has expired` ?

Reply:
I'll double-check with Fabio. This is the only message Kibana has and the idea was to be consistent when possible.

Reply:
I don't think we have a grace period anymore. AFAIK we have warnings before the expiration date, and no service after that. Could you please double check what `isLicenseExpiring()` assumes?

Reply:
We still have a grace period. We may not in the future, but we do now.

Reply:
_(sorry I cannot find the description of the full approach, so it may be already this)_

What about having two separate checks with two separate messages?

- 7 days before expiration — expiration: "will expire in N days"
- expiration — end of grace period: "has expired N days ago"

Reply:
I changed the behaviour to have 3 different messages:
 - 7 days before expiration — expiration: "will expire in N days"
- <1 days before expiration: "expires today"
- > expiration: "has expired N days ago"
If grace period is over this code should be irrelevant as the license is not active.

Reply:
Also, addad a test